### ISSUE TITLE
feat: Karpathy-inspired micro implementations for ix — micrograd.rs, nanogpt.rs, minbpe.rs

### ISSUE THREAD

**Issue by spareilleux** [GuitarAlchemist/Demerzel#121]
## Summary

Inspired by Karpathy's radical simplicity approach, create educational "micro" implementations in ix (Rust).

## Gap Analysis: Karpathy vs GuitarAlchemist

| Karpathy | Ours | Gap |
|---|---|---|
| micrograd (100 lines, autograd) | IxQL describes ML | No educational impl |
| nanoGPT (300 lines, train GPTs) | No equivalent | No LLM training |
| llm.c (C/CUDA training) | ix WGPU compute | No ML kernels |
| llama2.c (single-file inference) | tars API calls | No local inference |
| minbpe (BPE tokenizer) | IxQL embedding | No tokenizer impl |

### micro/ directory
- micro/micrograd.rs — scalar autograd, ~100 lines, teaches backprop
- micro/nanogpt.rs — transformer training, ~300 lines, teaches attention
- micro/minbpe.rs — BPE tokenizer, ~100 lines, teaches tokenization

### kernels/ directory (WGPU compute shaders)
- kernels/matmul.wgsl — matrix multiplication
- kernels/attention.wgsl — flash attention
- kernels/layernorm.wgsl — layer normalization
- kernels/softmax.wgsl — numerically stable softmax

### IxQL integration
```ixql
-- Train a nano GPT on Shakespeare
download("shakespeare") → tokenize(minbpe, vocab: 50257)
  → nanogpt(layers: 6, heads: 6, dim: 384)
  → train(epochs: 10, lr: 3e-4)
  → sample("To be or not to be")
```

### examples/ directory
Self-contained .rs files demonstrating each IxQL pattern — one file, one concept.

## Philosophy
"The simplest implementation that still works." — matches Manifesto principle #9 (factory of factories) applied to education: don't just describe ML, make it runnable in minimal code.

## Department
Computer Science + Data Visualization (for kernel profiling dashboards)

## References
- github.com/karpathy/micrograd
- github.com/karpathy/nanoGPT
- github.com/karpathy/llm.c
- grammars/sci-ml-pipelines.ebnf (IxQL)
- #103 (ix CLI)

**Comment by spareilleux** [comment]
🤖 **Demerzel Governance Notice**

This issue has been added to the [Demerzel Governance](https://github.com/orgs/GuitarAlchemist/projects/2) project board.

**Governance checklist:**
- [ ] Reviewed against [Asimov constitution](https://github.com/GuitarAlchemist/Demerzel/blob/master/constitutions/asimov.constitution.md) (Articles 0-5)
- [ ] Checked [default constitution](https://github.com/GuitarAlchemist/Demerzel/blob/master/constitutions/default.constitution.md) compliance (Articles 1-11)
- [ ] Risk assessed per [autonomous loop policy](https://github.com/GuitarAlchemist/Demerzel/blob/master/policies/autonomous-loop-policy.yaml)
- [ ] [Reconnaissance](https://github.com/GuitarAlchemist/Demerzel/blob/master/policies/reconnaissance-policy.yaml) completed if cross-repo

*Governed by Demerzel under the [governance mandate](https://github.com/GuitarAlchemist/Demerzel/blob/master/constitutions/demerzel-mandate.md).*
